contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -1099,9 +1099,9 @@ BEGIN
 	WHEN 'second' THEN
 		RETURN startdate OPERATOR(sys.+) make_interval(secs => num);
 	WHEN 'millisecond' THEN
-		RETURN startdate OPERATOR(sys.+) make_interval(secs => num * 0.001);
-	WHEN 'microsecond' THEN
-		RETURN startdate OPERATOR(sys.+) make_interval(secs => num * 0.000001);
+		RETURN startdate OPERATOR(sys.+) make_interval(secs => CAST(num AS DECIMAL(20,10)) * 0.001);
```

Comment:
`numeric` is preferred to decimal, and there's no reason to specify scale and precision. Just do `num::numeric`. FWIW, `pg_catalog.interval(num || ' millisecond')` would also work. It might be worth switching to that just to reduce the amount of code here. Also, `make_interval` should be schema-qualified, or use the `SET search_path = pg_catalog` option of CREATE FUNCTION.